packages/parser/src/language/timeline/timelineMatcher.ts

```diff
@@ -0,0 +1,28 @@
+import { CustomPatternMatcherFunc } from 'chevrotain';
+
+/**
+ * Matches a timeline section title
+ */
+export const timelineSectionTitleRegex = /section[\t ]+([^\n\r]+)/y;
+export const matchTimelineSectionTitle: CustomPatternMatcherFunc = (text, startOffset) => {
+  timelineSectionTitleRegex.lastIndex = startOffset;
+  return timelineSectionTitleRegex.exec(text);
+};
+
+/**
+ * Matches a timeline period title
+ */
+export const timelinePeriodTitleRegex = /([^\n\r:]+)/y;
+export const matchTimelinePeriodTitle: CustomPatternMatcherFunc = (text, startOffset) => {
+  timelinePeriodTitleRegex.lastIndex = startOffset;
+  return timelinePeriodTitleRegex.exec(text);
+};
+
+/**
+ * Matches a timeline period event
+ */
+export const timelinePeriodEventRegex = /:[\t ]+([^\n\r:]+)/y;
+export const matchTimelinePeriodEvent: CustomPatternMatcherFunc = (text, startOffset) => {
+  timelinePeriodEventRegex.lastIndex = startOffset;
+  return timelinePeriodEventRegex.exec(text);
+};
```

Comment:
I have updated it, but the last two test cases of `timeline` are failing. The issue is that the `startOffset` increased by a number that passed the `:`, so when trying to parse `TIMELINE_PERIOD_EVENT`, it doesn't find any `:`.

<https://github.com/mermaid-js/mermaid/blob/d06b44abb56cdfde34b9a5b8a9ae85de5acc214a/packages/parser/tests/timeline/syntax.test.ts#L55-L58>

`task1` parsed correctly, the `startOffset` passed the `: ` after `task1` and trying to parse `event1\n     task2: event2: event3` just like that.

Not really sure what the problem is, `:` isn't a hidden rule, the space has been ignored since it's a hidden rule.

Reply:
Can you check what your terminals actually return? And how does your token array look like? Assuming that the `TIMELINE_SECTION_TITLE` terminal only returns a match with `task1` the `startOffset` should stay at the `:` character.

Reply:
Are you still using your language defined in `Mermaid.langium`? Because that contains a `:` keyword that is probably the cause of this lexer behavior.

Reply:
> Are you still using your language defined in `Mermaid.langium`?

Yes, I wasn't really sure what and how should I change other files, I did what you suggested: [mermaidServiceRegistry.ts](https://github.com/mermaid-js/mermaid/blob/d06b44abb56cdfde34b9a5b8a9ae85de5acc214a/packages/parser/src/language/mermaid/mermaidServiceRegistry.ts), but not sure how to integrate other services, i.e, `ValueConverter`, `TokenBuilder`.

> Aside from that, you might want to reintegrate the regexes from the `timeline` language back into the `timeline.langium` file. Otherwise, the `longer_alt` property that is being set on keyword tokens doesn't work as expected.

I'll give it a try, I guess the `TIMELINE_SECTION_TITLE` would work fine since it has a keyword at the beginning, but the not sure about the others.

Reply:
> I'll give it a try, I guess the `TIMELINE_SECTION_TITLE` would work fine since it has a keyword at the beginning, but the not sure about the others.

The `TIMELINE_SECTION_TITLE` and `TIMELINE_PERIOD_EVENT` could be reintegrated back into `timeline.langium` (section title is reintegrated), but the `TIMELINE_PERIOD_TITLE` isn't because it would make some of the other rules `UNREACHABLE` when running `langium-cli`:

```
Token: ->ACC_DESCR<- can never be matched.
Because it appears AFTER the Token Type ->TIMELINE_PERIOD_TITLE<-in the lexer's definition.
See https://chevrotain.io/docs/guide/resolving_lexer_errors.html#UNREACHABLE-----------------------
Token: ->ACC_TITLE<- can never be matched.
Because it appears AFTER the Token Type ->TIMELINE_PERIOD_TITLE<-in the lexer's definition.
See https://chevrotain.io/docs/guide/resolving_lexer_errors.html#UNREACHABLE-----------------------
Token: ->TITLE<- can never be matched.
Because it appears AFTER the Token Type ->TIMELINE_PERIOD_TITLE<-in the lexer's definition.
See https://chevrotain.io/docs/guide/resolving_lexer_errors.html#UNREACHABLE-----------------------
```

Reply:
The main issue was resolved and the test cases are passing.
I think now I have separate the languages. I'll add regexes to the `.langium` files if they aren't failing.

Thanks for your help!